File: matlab/ch09/ex915.m
% levelcon.m
Kc1 = 5; Kc2 = 20; Kc3 = 50; t = [0:0.1:10]; num1 = Kc1; num2 = Kc2; num3 = Kc3;
den1 = [5 1+Kc1]; den2 = [5 1+Kc2]; den3 = [5 1+Kc3];
y1 = step(num1, den1, t); y2 = step(num2, den2, t); y3 = step(num3, den3, t);
plot(t, y1, ':', t, y2, '--', t, y3), xlabel('Time t(sec)'), ylabel('output y (t)'), legend('Kc = 5', 'Kc = 20', 'Kc = 50')

File: python/ch09/ex915.py
import control as ct
import matplotlib.pyplot as plt
import numpy as np

# 1. 설정값 및 시간 범위 정의
Kc_list = [5, 20, 50]
t = np.arange(0, 10.1, 0.1) # [0:0.1:10]
styles = [':', '--', '-']   # 매트랩의 선 스타일 대응
labels = ['Kc = 5', 'Kc = 20', 'Kc = 50']

# 2. 반복문을 이용한 시스템 정의 및 계단 응답 계산
for Kc, style, label in zip(Kc_list, styles, labels):
    num = [Kc]
    den = [5, 1 + Kc]
    
    # 시스템 생성
    sys = ct.tf(num, den)
    
    # 계단 응답 계산 (지정된 시간 t 사용)
    time, response = ct.step_response(sys, T=t)
    
    # 그래프 그리기
    plt.plot(time, response, style, label=label)

# 3. 그래프 꾸미기
plt.xlabel('Time t(sec)')
plt.ylabel('output y (t)')
plt.legend()
plt.grid(True)
plt.title('Step Responses for different Kc')
plt.show()